File: web/src/krate/kspace/store/dto.ts
import { KSpace } from "./po";

export type KSpaceQueryAllReq = object;

export type KSpaceQueryAllRsp = {
  kspaces: KSpace[];
};

export type KSpaceOverwriteReq = {
  kspace: KSpace;
};

export type KSpaceOverwriteRsp = object;

File: server/src/krate/kspace/dto.rs
use serde::{Deserialize, Serialize};

use super::KSpace;

#[derive(Debug, Clone, Serialize, Deserialize)]
pub(crate) struct KSpaceQueryAllReq {

}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub(crate) struct KSpaceQueryAllRsp {
    pub(crate) kspaces: Vec<KSpace>
}


#[derive(Debug, Clone, Serialize, Deserialize)]
pub(crate) struct KSpaceOverwriteReq {
    pub(crate) kspace: KSpace
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub(crate) struct KSpaceOverwriteRsp {}